tensorboard/webapp/app_routing/route_config.ts

```diff
@@ -249,20 +313,24 @@ export class RouteConfigs {
 
       this.configMatchers.push(matcher);
 
-      if (matcher.config.routeKind !== null) {
-        this.routeKindToConfigMatchers.set(matcher.config.routeKind, matcher);
+      if (matcher.config.routeType === RouteType.CONCRETE) {
+        this.routeKindToConfigMatchers.set(
```

Comment:
`routekindtoconfigmatchers` only ever stores CONCRETE config matchers, and L385 checks at runtime whether a value from the map is type CONCRETE.  Now that this PR introduces a convenient `ConcreteRouteConfigMatcher`, can we drop part of the L385 check and type the map as `routekindtoconfigmatchers: Map<RouteKind, ConcreteRouteConfigMatcher>`?